### PR TITLE
Batch completions V2 job

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

Batch completions revamp w/ vLLM v0.5.4 upgrade + OpenAI compatible request inputs. 

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

## Test Plan and Usage Guide

Tested w/ gemma 2 job (see vllm README for runnning locally)

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: clients/python/llmengine/completion.py [hunk 1]
```diff
@@ -480,6 +480,7 @@ def batch_create(
         data_parallelism: int = 1,
         max_runtime_sec: int = 24 * 3600,
         tool_config: Optional[ToolConfig] = None,
+        use_v2: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (yunfeng-scale) [reply clients/python/llmengine/completion.py:483]: what if we have v3 in the future?

[the remark this answers is not in the document]
Comment (dmchoiboi) [reply clients/python/llmengine/completion.py:483]: we could; it's almost the same API

[the remark this answers is not in the document]
Comment (dmchoiboi) [reply clients/python/llmengine/completion.py:483]: the intention is to remove this flag entirely and deprecate the v1 endpoint, but we need to roll out first

File: clients/python/setup.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 setup(
     name="scale-llm-engine",
     python_requires=">=3.7",
```
Comment (yunfeng-scale) [inline clients/python/setup.py:5]: update this to 3.8?

File: model-engine/model_engine_server/common/dtos/llms/batch_completion.py [hunk 3]
```diff
@@ -243,6 +265,26 @@ class BatchCompletionsJob(BaseModel):
 CreateBatchCompletionsV2Response: TypeAlias = BatchCompletionsJob
 
 
+class UpdateBatchCompletionsV2Request(BaseModel):
+    job_id: str = Field(description="ID of the batch completions job")
+    priority: Optional[int] = Field(
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/common/dtos/llms/batch_completion.py:270]: why would this be useful? wouldn't updating priority require a restart of the job?

File: model-engine/model_engine_server/inference/vllm/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+```bash
+VLLM_VERSION=0.5.4 bash build_and_upload_image.sh $ACCOUNT_ID $IMAGE_TAG {BUILD_TARGET=vllm|vllm_batch|vllm_batch_v2}
+```
+
+## Running locally
+
+### Endpoint
+
+1. Download model weights to `model_files`
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/vllm/README.md:18]: if not putting into justfile, maybe use some scripts?

File: model-engine/model_engine_server/inference/vllm/vllm_batch.py [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    ] = []
+    for request in requests:
+        if isinstance(request, CompletionRequest):
+            results_generators.append(openai_serving_completion.create_completion(request))
+        elif isinstance(request, ChatCompletionRequest):
+            results_generators.append(openai_serving_chat.create_chat_completion(request))
+        else:
+            assert_never(request)
+
+    results_generator = await_coroutines(*results_generators)
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/vllm/vllm_batch.py:169]: does this guarantee ordering?

File: model-engine/model_engine_server/inference/vllm/vllm_batch.py [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+            data = json.load(f)
+            try:
+                return CreateBatchCompletionsV1RequestContent.model_validate(data)
+            except ValidationError:
+                model = get_model_name(request.model_cfg)
+                if not isinstance(data, List) or len(data) == 0:
+                    raise ValueError("Input data must be a list of completion requests")
+
+                sample = data[0]
+                # Kinda jank, but we first serialize to FilteredCompletionV2Request before serializing to
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/vllm/vllm_batch.py:258]: i think it's okay but may consider adding a type so we readily know what the type is?

File: model-engine/model_engine_server/inference/vllm/vllm_batch.py (hunk 5 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/inference/vllm/vllm_batch.py:169]: the coroutines don't complete in order, but the function itself wraps each generator with the corresponding index in the sequence/list so the output order is fixed

File: model-engine/model_engine_server/inference/vllm/vllm_batch.py (hunk 6 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/inference/vllm/vllm_batch.py:258]: The type is FilteredCompletionV2Request that we cast to vLLMs CompletionRequest. I think vLLMs pydantic spec is too strict, but that's out of our control

File: model-engine/model_engine_server/common/dtos/llms/batch_completion.py (hunk 3 above)
Comment (dmchoiboi) [reply model-engine/model_engine_server/common/dtos/llms/batch_completion.py:270]: this would only apply for jobs in queue

**Review by yunfeng-scale — APPROVED** [review]

**Pull request merged**